tailnet/configmaps.go

```diff
@@ -56,10 +56,11 @@ type phased struct {
 
 type configMaps struct {
 	phased
-	netmapDirty  bool
-	derpMapDirty bool
-	filterDirty  bool
-	closing      bool
+	netmapDirty           bool
+	derpMapDirty          bool
+	filterDirty           bool
+	closing               bool
+	waitReadyForHandshake bool
```

Comment:
Making this state scoped to the `configMaps` means that, essentially, we're building a toggle that says:

T: act like a client
F: act like an agent

But, the protocol is deliberately built to avoid those categorizations as we were already straining against them.  Instead the model is peers and tunnels. Although this isn't allowed at the higher layers, at the protocol level a peer could be the source for one tunnel, the destination for another. So, we want the _source of the tunnel_ to wait for the READY signal. The state is scoped per tunnel (or from the perspective of `configMaps`, per peer).

So, when the higher layer is setting up a tunnel, it should make a call into `Conn` which calls into `configMaps` to set up the per-peer state that we are the source of the tunnel and need to wait until we get both the NODE and the READY state.